server/shellquote/shellquote_test.go

```diff
@@ -0,0 +1,58 @@
+// Based on https://github.com/kballard/go-shellquote
+
+package shellquote
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestSimpleSplit(t *testing.T) {
+	for _, elem := range simpleSplitTest {
+		output, err := Split(elem.input)
+		if err != nil {
+			t.Errorf("Input %q, got error %#v", elem.input, err)
+		} else if !reflect.DeepEqual(output, elem.output) {
+			t.Errorf("Input %q, got %q, expected %q", elem.input, output, elem.output)
+		}
+	}
+}
+
+func TestErrorSplit(t *testing.T) {
+	for _, elem := range errorSplitTest {
+		_, err := Split(elem.input)
+		if !errors.Is(err, elem.error) {
+			t.Errorf("Input %q, got error %#v, expected error %#v", elem.input, err, elem.error)
+		}
+	}
+}
+
+var simpleSplitTest = []struct {
```

Comment:
Nit: While we're at it, add a test for flags like `--foo bar` or with characters that might be on server private keys like `ozVjMTacNzulUo7E9dhL2TRGKljszQWszh9HYpRcI0c=`?